release/lightning_tests/workloads/test_tuner.py

```diff
@@ -0,0 +1,150 @@
+
+    def validation_epoch_end(self, outputs):
+        avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
+        avg_acc = torch.stack([x["val_accuracy"] for x in outputs]).mean()
+        self.log("ptl/val_loss", avg_loss.item(), sync_dist=True)
+        self.log("ptl/val_accuracy", avg_acc.item(), sync_dist=True)
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
+        return optimizer
+
+
+class MNISTDataModule(pl.LightningDataModule):
+    def __init__(self, batch_size=100):
+        super().__init__()
+        self.data_dir = os.getcwd()
+        self.batch_size = batch_size
+        self.transform = transforms.Compose(
+            [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+        )
+
+    def setup(self, stage=None):
+        # split data into train and val sets
+        if stage == "fit" or stage is None:
+            mnist = MNIST(
+                self.data_dir, train=True, download=True, transform=self.transform
+            )
+            self.mnist_train, self.mnist_val = random_split(mnist, [55000, 5000])
+
+        # assign test set for use in dataloader(s)
+        if stage == "test" or stage is None:
+            self.mnist_test = MNIST(
+                self.data_dir, train=False, download=True, transform=self.transform
+            )
+
+    def train_dataloader(self):
+        return DataLoader(self.mnist_train, batch_size=self.batch_size, num_workers=4)
+
+    def val_dataloader(self):
+        return DataLoader(self.mnist_val, batch_size=self.batch_size, num_workers=4)
+
+    def test_dataloader(self):
+        return DataLoader(self.mnist_test, batch_size=self.batch_size, num_workers=4)
+
+
+if __name__ == "__main__":
+    start = time.time()
+
+    lightning_config = (
+        LightningConfigBuilder()
+        .module(
+            MNISTClassifier,
+            feature_dim=tune.choice([64, 128]),
+            lr=tune.grid_search([0.01, 0.001]),
+        )
+        .trainer(
+            max_epochs=5,
+            accelerator="gpu",
+            logger=CSVLogger("logs", name="my_exp_name"),
+        )
+        .fit_params(datamodule=MNISTDataModule(batch_size=200))
+        .checkpointing(monitor="ptl/val_accuracy", mode="max")
+        .build()
+    )
+
+    scaling_config = ScalingConfig(
+        num_workers=3, use_gpu=True, resources_per_worker={"CPU": 1, "GPU": 1}
+    )
+
+    lightning_trainer = LightningTrainer(
+        scaling_config=scaling_config,
+    )
+
+    tuner = tune.Tuner(
+        lightning_trainer,
+        param_space={"lightning_config": lightning_config},
+        run_config=ray.air.RunConfig(
+            name="release-tuner-test",
+            verbose=2,
+            checkpoint_config=CheckpointConfig(
+                num_to_keep=1,
+                checkpoint_score_attribute="ptl/val_accuracy",
+                checkpoint_score_order="max",
```

Comment:
Isn't this a bit dangerous 😅

Reply:
Oh wait, have we resolved this num_to_keep=1 problem?

Reply:
I don't think so